torch/fx/symbolic_trace.py

```diff
@@ -307,13 +307,140 @@ def forward(*args, **kwargs):
             # Seems to be a mypy limitation: https://github.com/python/mypy/issues/2427
             torch.nn.Module.__getattr__ = module_getattr_wrapper  # type: ignore
             torch.nn.Module.__call__ = module_call_wrapper
+
+            orig_fns = _patch_wrapped_functions()
+
             self.create_node('output', 'output', (self.create_arg(fn(*args)),), {},
                              type_expr=fn.__annotations__.get('return', None))
         finally:
+            _unpatch_wrapped_functions(orig_fns)
             torch.nn.Module.__call__ = orig_call
             torch.nn.Module.__getattr__ = orig_getattr  # type: ignore
         return self.graph
 
+class _GlobalDictWrapper:
```

Comment:
I don't think it is necessary to hash the global dictionaries. The patch table can just be:
```
_wrapped_fn_patch_list : List[Tuple[dict, str]]
```

Grouping the patches by global dictionary really doesn't save any time anywhere. True you deduplicate things if someone wrapped something twice but that shouldn't actually happen in a way where it would seriously affect performance. Wrapping an already wrapped function would just be a no-op.